ledger/accountdb_test.go

```diff
@@ -2131,6 +2131,179 @@ func TestResourcesDataSetData(t *testing.T) {
 	}
 }
 
+// TestResourceDataRoundtripConversion ensures that basics.AppLocalState, basics.AppParams,
+// basics.AssetHolding, and basics.AssetParams can be converted to resourcesData and back without
+// losing any data. It uses reflection to be sure that no new fields are omitted.
+//
+// In other words, this test makes sure any new fields in basics.AppLocalState, basics.AppParams,
+// basics.AssetHolding, or basics.AssetParam also get added to resourcesData.
+func TestResourceDataRoundtripConversion(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	t.Run("basics.AppLocalState", func(t *testing.T) {
+		t.Parallel()
+		for i := 0; i < 1000; i++ {
+			randObj, _ := protocol.RandomizeObject(&basics.AppLocalState{})
+			basicsAppLocalState := *randObj.(*basics.AppLocalState)
+
+			var data resourcesData
+			data.SetAppLocalState(basicsAppLocalState)
+			roundTripAppLocalState := data.GetAppLocalState()
+
+			require.Equal(t, basicsAppLocalState, roundTripAppLocalState)
+		}
+	})
+
+	t.Run("basics.AppParams", func(t *testing.T) {
+		t.Parallel()
+		for i := 0; i < 1000; i++ {
+			randObj, _ := protocol.RandomizeObject(&basics.AppParams{})
+			basicsAppParams := *randObj.(*basics.AppParams)
+
+			var data resourcesData
+			data.SetAppParams(basicsAppParams, false)
+			roundTripAppParams := data.GetAppParams()
+
+			require.Equal(t, basicsAppParams, roundTripAppParams)
+		}
+	})
+
+	t.Run("basics.AssetHolding", func(t *testing.T) {
+		t.Parallel()
+		for i := 0; i < 1000; i++ {
+			randObj, _ := protocol.RandomizeObject(&basics.AssetHolding{})
+			basicsAssetHolding := *randObj.(*basics.AssetHolding)
+
+			var data resourcesData
+			data.SetAssetHolding(basicsAssetHolding)
+			roundTripAssetHolding := data.GetAssetHolding()
+
+			require.Equal(t, basicsAssetHolding, roundTripAssetHolding)
+		}
+	})
+
+	t.Run("basics.AssetParams", func(t *testing.T) {
+		t.Parallel()
+		for i := 0; i < 1000; i++ {
+			randObj, _ := protocol.RandomizeObject(&basics.AssetParams{})
+			basicsAssetParams := *randObj.(*basics.AssetParams)
+
+			var data resourcesData
+			data.SetAssetParams(basicsAssetParams, false)
+			roundTripAssetParams := data.GetAssetParams()
+
+			require.Equal(t, basicsAssetParams, roundTripAssetParams)
+		}
+	})
+}
+
+// TestBaseAccountDataRoundtripConversion ensures that baseAccountData can be converted to
+// ledgercore.AccountData and basics.AccountData and back without losing any data. It uses
+// reflection to be sure that no new fields are omitted.
+//
+// In other words, this test makes sure any new fields in baseAccountData also get added to
+// ledgercore.AccountData and basics.AccountData. You should add a manual override in this test if
+// the field really only belongs in baseAccountData.
+func TestBaseAccountDataRoundtripConversion(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	t.Run("ledercore.AccountData", func(t *testing.T) {
```

Comment:
@jasonpaulos Minor + optional typo _suggestion_.

There's also a variable (`ledercoreAccount`) with the same class of typo.

```suggestion
	t.Run("ledgercore.AccountData", func(t *testing.T) {
```

Reply:
Thanks, fixed in 8dc7b4f